### PR TITLE
Diff null sources

### PR DESCRIPTION
**Pull request by hedgar2017**
Resolves the [issue](https://github.com/foundpatterns/torchbear/issues/203)

### CODE REVIEW

File: src/bindings/text/diff.rs [hunk 1]
```diff
@@ -53,8 +58,23 @@ pub fn init(lua: &Lua) -> crate::Result<()> {
...
+            )));
+        }
+
+        let left = if left != NULL_SOURCE {
+            read_file(&left).map_err(LuaError::external)?
+        } else {
+            Vec::new()
+        };
+        let right = if right != NULL_SOURCE {
+            Vec::new()
```
Comment (dariusc93) [inline src/bindings/text/diff.rs]: Should this be `read_file(&right).map_err(LuaError::external)?`?

**Comment by dariusc93** [comment]
What you should do to reduce repeated code is to put 
```
#[cfg(target_os = "windows")]
const NULL_SOURCE: &str = "nul";
#[cfg(target_os = "linux")]
const NULL_SOURCE: &str = "/dev/null";
```
in `text/mod.rs` then do `use crate::bindings::text::NULL_SOURCE` in any code, or maybe put them in one of the crates this crate uses (eg patch?) and include it. Same with PR #258 and maybe #255 ?

File: src/bindings/text/diff.rs (hunk 1 above)
Comment (hedgar2017) [reply src/bindings/text/diff.rs]: Fixed

**Comment by hedgar2017** [comment]
Merged with PR #258

**Pull request closed without merging**